onnxruntime/core/session/inference_session.h

```diff
@@ -89,6 +95,10 @@ struct SessionOptions {
 
   // How many threads in the session thread pool.
   int session_thread_pool_size = -1;
+
+  // For models with free input dimensions (most commonly batch size), specifies a set of values to override those
+  // free dimensions with, keyed by dimension denotation.
+  std::vector<FreeDimensionOverride> free_dimension_overrides;
```

Comment:
Should we also expose this to the C API?

Reply:
Yes, I'll be doing this shortly once this is merged. In fact, this has to be exposed in the C API for the DML integration to be successful or else the models won't run with the same performance.

Reply:
Maybe it's also useful for the other execution providers. Because when they are symbolic shapes, some part of the memory planner doesn't work.